risc0/zkp/src/prove/soundness.rs

```diff
@@ -0,0 +1,292 @@
+use crate::{hal::Hal, taps::TapSet, FRI_FOLD, FRI_MIN_DEGREE, INV_RATE};
+use risc0_core::field::{baby_bear, ExtElem};
+
```

Comment:
Would be helpful to add a code comment giving a high-level orientation to what this file contains. something like: 

A soundness calculator for the RISC Zero STARK protocol that secures the RISC Zero zkVM. 

Soundness for STARK protocols can be analyzed under a number of different cryptographic assumptions. RISC Zero targets 100 bits of conjectured soundness, using the Toy Problem Conjecture. 

For completeness, we also include analysis in three other regimes:
- Conjectured soundness using Conjecture 8.4 from [Proximity Gaps] and Conjecture 2.3 from [DEEP-FRI]
- Proven soundness in the list-decoding regime
- Proven soundness in the unique-decoding regime